index.js

```diff
@@ -21,8 +21,8 @@ module.exports = function (filename, opts) {
   var since = Obv()
 
   var waiting = []
-  var waitingDrain = {} // blockIndex -> []
-  var blocksToBeWritten = new Map() // blockIndex -> { block, fileOffset }
+  const waitingDrain = new Map() // blockIndex -> []
```

Comment:
I changed this one from object to Map because there were some `delete`s. Overall we should absolutely avoid doing `delete obj.foo` on hot path code, because it's quite slow.